Sources/Core/Microsoft.StreamProcessing/Aggregates/TopKState.cs

```diff
@@ -0,0 +1,236 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.StreamProcessing.Aggregates
+{
+    /// <summary>
+    /// State used by TopK Aggregate
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public interface ITopKState<T>
```

Comment:
>ITopKState [](start = 21, length = 10)

This will wrap every aggregation operation a virtual function call. I wonder if it would be better to consolidate them with a single class and differentiate behavior at runtime like the Min/Max state, especially if we optimize the simple top k state as well so it will also have a comparer, etc.